### PR TITLE
Issue 29

### PR DESCRIPTION
**Pull request by rendel**

### CODE REVIEW

File: src/main/java/solutions/siren/join/SirenJoinPlugin.java [hunk 1]
```diff
@@ -89,4 +92,9 @@ public String description() {
     return "SIREn plugin that adds join capabilities to Elasticsearch";
   }
 
+  @Override
+  public Settings additionalSettings() {
+    return Settings.builder().put(IndexCacheModule.QUERY_CACHE_EVERYTHING, true).build();
```
Comment (scampi) [inline src/main/java/solutions/siren/join/SirenJoinPlugin.java]: So we can remove the option in the elasticsearch.yml of the kibi distribution ?

File: src/main/java/solutions/siren/join/action/terms/collector/IntegerTermsSet.java [hunk 2]
```diff
@@ -115,8 +115,9 @@ public void writeTo(StreamOutput out) throws IOException {
   @Override
   public BytesRef writeToBytes() {
     long start = System.nanoTime();
-
     int size = set.size();
+
+//    this.adjustBreaker(HEADER_SIZE + 5 * size);
```
Comment (scampi) [inline src/main/java/solutions/siren/join/action/terms/collector/IntegerTermsSet.java]: commented code

File: src/main/java/solutions/siren/join/action/terms/collector/IntegerTermsSet.java [hunk 3]
```diff
@@ -162,4 +163,58 @@ private void readFromBytes(BytesRef bytesRef) {
...
+    if (set != null) {
+      set.release();
+    }
+  }
+
+  /**
+   * A {@link IntHashSet} integrated with the {@link CircuitBreaker}. It will adjust the circuit breaker
+   * for every new call to {@link #allocateBuffers(int)}.
+   * <p>
+   * This set must not be reused after a call to {@link #release()}.
```
Comment (scampi) [inline src/main/java/solutions/siren/join/action/terms/collector/IntegerTermsSet.java]: Should we put a flag `isreleased` to check that ? Maybe in a future version

File: src/main/java/solutions/siren/join/action/terms/collector/LongTermsSet.java [hunk 4]
```diff
@@ -121,6 +127,7 @@ public BytesRef writeToBytes() {
     long start = System.nanoTime();
     int size = set.size();
 
+//    this.adjustBreaker(HEADER_SIZE + 8 * size);
```
Comment (scampi) [inline src/main/java/solutions/siren/join/action/terms/collector/LongTermsSet.java]: commented code

File: src/main/java/solutions/siren/join/action/terms/collector/IntegerTermsSet.java (hunk 3 above)
Comment (rendel) [inline src/main/java/solutions/siren/join/action/terms/collector/IntegerTermsSet.java]: this is internal code, and these classes are not meant to be reused elsewhere at the moment. If people use them for something else, it is a their own risk.

File: src/main/java/solutions/siren/join/action/terms/collector/IntegerTermsSet.java (hunk 2 above)
Comment (rendel) [inline src/main/java/solutions/siren/join/action/terms/collector/IntegerTermsSet.java]: ok

**Comment by scampi** [comment]
:+1:

File: src/main/java/solutions/siren/join/SirenJoinPlugin.java (hunk 1 above)
Comment (rendel) [inline src/main/java/solutions/siren/join/SirenJoinPlugin.java]: Not entirely sure yet, we will have to do some test.

File: src/main/java/solutions/siren/join/action/terms/collector/LongTermsSet.java (hunk 4 above)
Comment (rendel) [inline src/main/java/solutions/siren/join/action/terms/collector/LongTermsSet.java]: ok

**Comment by rendel** [comment]
Thanks stephane.

**Pull request merged**